src/human_encoding/mod.rs

```diff
@@ -175,11 +175,13 @@ impl<J: Jet> Forest<J> {
                 let arrow_str = format!(": {} -> {}", arrow.source, arrow.target).replace('×', "*"); // for human-readable encoding stick with ASCII
 
                 // All witnesses have the same CMR so don't bother printing it
-                let cmr_str = if let node::Inner::Witness(..) = node.inner() {
+                let cmr_str = /*if let node::Inner::Witness(..) = node.inner() {
                     String::new()
-                } else {
-                    format!("-- cmr {:.8}...", node.cmr())
-                };
+                } else { */
+                    format!("-- cmr {}...", node.cmr())
+                    + &format!("  -- imr {}...", node.imr().as_ref().map(ToString::to_string).unwrap_or("???".to_string()))
+//                };
+;
```

Comment:
04f8ffad7e92ce024363816b3b13c7d457d29c15: We probably want to remove the special case for witness entirely.

Reply:
Oops, this code just shouldn't be there. *Or* we should step back and consider whether we want to add CMRs, IMRs, or both (and how). Right now it's an ad-hoc choice based on what I currently needed to generate blog posts.

Reply:
Let's fix this.

Reply:
I think `malleable` is misleading. How about `[undetermined]`? Or even `[undetermined at commitment time]` since we've got a lot of room for characters here.